src/Features/Core/Portable/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -45,6 +45,14 @@ internal abstract partial class AbstractInheritanceMarginService
             ImmutableArray<(SymbolKey symbolKey, int lineNumber)> symbolKeyAndLineNumbers,
             CancellationToken cancellationToken)
         {
+            // If we're starting from a document, use it to go to a frozen partial version of it to lower the amount of
+            // work we need to do running source generators or producing skeleton references.
+            if (documentForGlobalImports != null)
+            {
+                documentForGlobalImports = documentForGlobalImports.WithFrozenPartialSemantics(cancellationToken);
```

Comment:
This is already a document with frozen partial semantics (per caller)

Reply:
note that for classification what we do is actually send a boolean over to OOP saying taht it should be frozen-partial again on its end.

this is needed as frozen partial effectively means 'capture whatever completed computations are available'.  but hte set of completed computations is different on host/oop.  so if OOP hasn't performed some comptuations we don't want it to now pay that price to try to get it's computations in sync with the host.

Reply:
> note that for classification what we do is actually send a boolean over to OOP saying taht it should be frozen-partial again on its end.

We do this for several features, and should match it here.

Reply:
will do.